hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -898,6 +899,38 @@ boolean tryLock(K key, long time, TimeUnit timeunit)
      */
     boolean removeEntryListener(String id);
 
+
+    /**
+     * Adds a MapPartitionLostListener.
+     * <p/>
+     * The addPartitionLostListener returns a register-id. This id is needed to remove the MapPartitionLostListener using the
+     * {@link #removePartitionLostListener(String)} method.
+     * <p/>
+     * There is no check for duplicate registrations, so if you register the listener twice, it will get events twice.
+     * IMPORTANT: Please @see com.hazelcast.partition.PartitionLostListener for weaknesses
+     * IMPORTANT: Listeners registered from HazelcastClient are assigned to nodes randomly. When a node to which
```

Comment:
I just want to inform user that some events may be lost if the listener is registered via client. I will do it with less details.